test/nofiles.test.js

```diff
@@ -0,0 +1,90 @@
+const stylelint = require('stylelint')
+const path = require('path')
+
+const code = `import styled from 'styled-components';
+const StyledTable = styled(StyledTableBase)\`
+color: red;
+\``
+const processor = path.join(__dirname, '../lib/index.js')
+const rules = {
+  indentation: 2
+}
+
+describe('no files', () => {
+  let codeFilename
+  let data
+
+  // NOTE beforeEach() runs _after_ the beforeAll() hooks of the describe() blocks
+  beforeEach(done => {
+    stylelint
+      .lint({
+        code,
+        codeFilename,
+        syntax: 'scss',
+        config: {
+          processors: [processor],
+          rules
+        }
+      })
+      .then(result => {
+        data = result
+        done()
+      })
+      .catch(err => {
+        // eslint-disable-next-line
+        console.log(err)
+        data = err
+        done()
+      })
+  })
+
+  describe('no codeFilename', () => {
+    beforeAll(() => {})
+
+    it('should have one result', () => {
+      expect(data.results.length).toEqual(1)
+    })
+
+    it('should use the right file', () => {
+      expect(data.results[0].source).toEqual('<input css 2>')
```

Comment:
This is more curiosity, as the test obviously passes. The reason the id here is 2 is because the `beforeEach` runs before each of these and therefore for each `it` call, a new linting occurs which increments the id?

Reply:
Exactly as you thought. If it is hard to understand, shall we add comments here?

Reply:
Yeah sure, why not? :). And then I'll merge it in right away.